snafu/benchmarks/dnsperf/dnsperf.py

```diff
@@ -0,0 +1,234 @@
+        ConfigArgument("--network-policy", dest="networkpolicy", env_var="networkpolicy"),
+        ConfigArgument("--pod-id", dest="pod_id", env_var="my_pod_idx", default=None),
+        ConfigArgument("--steps", dest="steps", default=5),
+    )
+
+    def setup(self) -> bool:
+        """Setup dnsperf benchmark."""
+        self.logger.info("Setting up dnsperf benchmark.")
+        self.config.parse_args()
+
+        template_path = Path(__file__).with_name("dnsperf-template.xml")
+        with template_path.open() as template_f:
+            self.output_template = template_f.read()
+
+        if self.config.pod_id:
+            self.tmp_filepath += f"-{self.pod_id}"
+        if check_file(self.tmp_filepath):
+            self.logger.critical((f"Temporary file located at {self.tmp_filepath} " "already exists."))
+            return False
+        try:
+            tmp_file = open(self.tmp_filepath, "x")
+            tmp_file.close()
+        except Exception as e:
+            self.logger.critical(
+                f"Unable to create temporary file at {self.tmp_filepath}: {e}", exc_info=True
+            )
+            return False
+        else:
+            self.logger.info(f"Successfully create temporary file: {self.tmp_filepath}")
+        return True
+
+    def cleanup(self) -> bool:
+        """Clean up artifacts from the dnsperf benchmark."""
+        self.logger.info("Cleaning up dnsperf benchmark.")
+        try:
+            os.remove(self.tmp_filepath)
+        except Exception as e:
+            self.logger.critical(f"Unable to remove temporary file {self.tmp_filepath}: {e}", exc_info=True)
+            return False
+        self.logger.info(f"Successfully removed temp file {self.tmp_filepath}")
+        return True
+
+    def _one_trial(self, load=None) -> BenchmarkResult:
+        cmd = [
+            "dnsperf",
+            "-v",  # print latency information for each query
+            "-s",
+            self.config.address,
+            "-p",
+            self.config.port,
+            "-d",
+            self.config.query_filepath,
+            "-c",
+            self.config.clients,
+            "-l",
+            self.config.time_limit,
+            "-m",
+            self.config.transport_mode,
+        ]
+
+        if load:
+            cmd = [*cmd, "-Q", str(load)]
+
+        sample: ProcessSample = sample_process(
+            cmd, self.logger,
+        )
+
+        if not sample.success:
+            self.logger.critical(f"dnsperf failed to complete! Got results: {sample}\n")
+            return
+        elif sample.successful.stdout is None:
+            self.logger.critical(
+                ("dnsperf ran successfully, but did not get output on stdout.\n" f"Got results: {sample}\n")
+            )
+            return
+
+        stdout: DnsperfStdout = self.parse_stdout(sample.successful.stdout)
+        cfg: DnsperfConfig = DnsperfConfig.new(stdout, self.config, load=load)
+        # self.logger.info(stdout)
```

Comment:
Should this info always logged to the terminal, or only when the user asks for verbose output?

Reply:
I'm voting for verbose output, that's my take